### PR TITLE
collection : be more permissive with dates format

### PR DESCRIPTION
**Pull request by AlicVB**
allow to use uncomplete date-time, like 2014 or 2014:05 etc...
only the year part is now mandatory

### CODE REVIEW

File: src/common/collection.c [hunk 1]
```diff
@@ -607,6 +607,66 @@ void dt_collection_split_operator_number(const gchar *input, char **number1, cha
...
+  {
+    if(!strptime(input, "%Y:%m:%d %H:%M", &tm1)) return NULL;
+  }
+  else
+  {
+    if(!strptime(input, "%Y:%m:%d %H:%M:%S", &tm1)) return NULL;
+  }
+
+  // we return the created date
+  char *ret = (char *)calloc(20, sizeof(char));
```
Comment (LebedevRI) [inline src/common/collection.c]: `g_malloc0_n()`, because you use `dt_util_dstrcat()`, which uses `g_realloc()`

Comment (AlicVB) [inline src/common/collection.c]: fixed

**Comment by LebedevRI** [comment]
Also a tooltip in collect.c needs changing
https://github.com/darktable-org/darktable/blob/0b74cbd7a448073ab849730c24f0577772d8375c/src/libs/collect.c#L1255

**Comment by AlicVB** [comment]
tooltip fixed. Thks for the review

**Comment by LebedevRI** [comment]
Seems ok to me.

**Pull request merged**